packages/background-charm-service/src/worker.ts

```diff
@@ -127,6 +130,31 @@ async function cleanup(): Promise<void> {
   initialized = false;
 }
 
+async function isInCharmList(
```

Comment:
this feels like it could be part of the charm manager? it looks like this is triggered from runCharm which has an active manager for the space?